### PR TITLE
Project/Dataset export import API alignment

### PR DESCRIPTION
**Pull request by igor-davidyuk**
Introduce methods for importing and exporting project and dataset snapshots following the web UI functionality.

### CODE REVIEW

**Review by ljcornel — COMMENTED** [review]
Great work, thanks a lot! 

Aside from a couple of minor remarks, one thing I'm still missing is the documentation for the new methods. I'm not sure what the best place to add this is, but I'd like to see a short list of the methods we have now for up/download, along with a short description. I think having an overview like that somewhere would help users to select the method appropriate for their use case.

Maybe we can add it to the import/export module __init__? That way it should be easy to include it in the sphinx documentation as well.

File: geti_sdk/import_export/tus_uploader.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        self.tus_resumable_version = self._get_tus_reumable_version()
+        self.offset = 0
+        self.upload_url = None
+        self.chunk_size = chunk_size
+        self.retries = retries
+        self.request = None
+        self._retried = 0
+        self.retry_delay = retry_delay
+
+    def _get_tus_reumable_version(self):
```
[the remark this answers is not in the document]
Comment (igor-davidyuk) [reply geti_sdk/import_export/tus_uploader.py:59]: fixed

File: geti_sdk/import_export/import_export_module.py [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+    ) -> Project:
+        """
+        Import a dataset as a new project to the Geti Platform.
+
+        :param filepath: The path to the dataset archive.
+        :param project_name: The name of the new project.
+        :param project_type: The type of the new project. Provide one of
+            [classification, classification_hierarchical, detection, segmentation,
+            instance_segmentation, anomaly_classification, anomaly_detection, anomaly_segmentation,
+            detection_oriented, detection_classification, detection_segmentation]
```
[the remark this answers is not in the document]
Comment (igor-davidyuk) [reply geti_sdk/import_export/import_export_module.py:369]: done!

File: notebooks/009_download_and_upload_project.ipynb [hunk 3]
```diff
@@ -134,7 +134,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "uploaded_project = geti.upload_project(\n",
+    "uploaded_project = geti.upload_project_Data(\n",
```
[the remark this answers is not in the document]
Comment (igor-davidyuk) [reply notebooks/009_download_and_upload_project.ipynb:137]: fixed

**Comment by igor-davidyuk** [comment]
> Maybe we can add it to the import/export module **init**? That way it should be easy to include it in the sphinx documentation as well.

The IE module does not expose these methods. Instead, I added the description for all the methods to the `Geti` module docstring.

**Review by ljcornel — APPROVED** [review]

**Pull request merged**